File: app/src/main/java/jp/ttanaka330/androidlearning/di/AppModule.java
package jp.ttanaka330.androidlearning.di;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import jp.ttanaka330.androidlearning.presentation.realm.RealmDatabase;

@Module
class AppModule {

    @Singleton
    @Provides
    RealmDatabase provideDatabase() {
        return new RealmDatabase();
    }

}

File: app/src/main/kotlin/jp/ttanaka330/androidlearning/di/AppModule.kt
package jp.ttanaka330.androidlearning.di

import dagger.Module
import dagger.Provides
import okhttp3.Interceptor
import okhttp3.OkHttpClient
import javax.inject.Singleton

@Module(includes = [BuildTypeAppModule::class])
internal class AppModule {

    @Singleton
    @Provides
    fun provideHttpClient(
        loggingInterceptors: Set<@JvmSuppressWildcards Interceptor>
    ): OkHttpClient = OkHttpClient.Builder().apply {
        loggingInterceptors.forEach { addInterceptor(it) }
    }.build()
}